### PR TITLE
feat/py312-compatibility

### PR DESCRIPTION
**Pull request by billy-doyle**
As discussed in multiple issues, but notably https://github.com/mingrammer/diagrams/issues/832#issuecomment-2329529817, remove typed-ast for 3.12 compatibility

### CODE REVIEW

**Comment by billy-doyle** [comment]
can't request a review, tagging @JelleZijlstra

**Comment by billy-doyle** [comment]
also have confirmed this works on x86 debian image and my m1 mac, probs would be worthwhile to add in some tests tho

**Comment by JelleZijlstra** [comment]
The change looks good; the library doesn't seem to use typed-ast anywhere.

Could you also update tests.yml to run tests on 3.12 too? And maybe 3.13-dev as well while we're here.

**Comment by billy-doyle** [comment]
yup taking a look here

**Comment by JelleZijlstra** [comment]
No CI again here, let me close and reopen to see if that helps

**Comment by billy-doyle** [comment]
name change in the file :)

**Comment by billy-doyle** [comment]
lmk if ya want a rebase or anything

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -5,20 +5,20 @@ on:
     branches:
       - master
     paths:
-      - ".github/workflows/test.yml"
+      - ".github/workflows/tests.yml"
```
Comment (JelleZijlstra) [inline .github/workflows/tests.yml:8]: Oh good catch

**Review by JelleZijlstra — APPROVED** [review]

**Pull request merged**